tests/distributed/test_distributed_sampling.py

```diff
@@ -1858,6 +1859,92 @@ def test_local_sampling_heterograph(num_parts, use_graphbolt, prob_or_mask):
                 )
 
 
+def check_mask_hetero_sampling_gb(tmpdir, num_server, use_graphbolt=True):
+    def create_hetero_graph(dense=False, empty=False):
+        num_nodes = {"n1": 210, "n2": 200, "n3": 220, "n4": 230}
+        etypes = [("n1", "r12", "n2"), ("n2", "r23", "n3"), ("n3", "r34", "n4")]
+        edges = {}
+        random.seed(42)
+        for etype in etypes:
+            src_ntype, _, dst_ntype = etype
+            arr = spsp.random(
+                num_nodes[src_ntype] - 10 if empty else num_nodes[src_ntype],
+                num_nodes[dst_ntype] - 10 if empty else num_nodes[dst_ntype],
+                density=0.1,
+                format="coo",
+                random_state=100,
+            )
+            edges[etype] = (arr.row, arr.col)
+        g = dgl.heterograph(edges, num_nodes)
+
+        return g
+
+    generate_ip_config("rpc_ip_config.txt", num_server, num_server)
+
+    g = create_hetero_graph()
+    eids = torch.randperm(g.num_edges("r34"))[:10]
+    mask = torch.zeros(g.num_edges("r34"), dtype=torch.bool)
+    mask[eids] = True
+
+    num_parts = num_server
+
+    orig_nid_map, orig_eid_map = partition_graph(
+        g,
+        "test_sampling",
+        num_parts,
+        tmpdir,
+        num_hops=1,
+        part_method="metis",
+        return_mapping=True,
+        use_graphbolt=use_graphbolt,
+        store_eids=True,
+    )
+
+    part_config = tmpdir / "test_sampling.json"
+
+    pserver_list = []
+    ctx = mp.get_context("spawn")
+    for i in range(num_server):
+        p = ctx.Process(
+            target=start_server,
+            args=(
+                i,
+                tmpdir,
+                num_server > 1,
+                "test_sampling",
+                ["csc", "coo"],
+                True,
+            ),
+        )
+        p.start()
+        time.sleep(1)
+        pserver_list.append(p)
+
+    dgl.distributed.initialize("rpc_ip_config.txt", use_graphbolt=True)
+    dist_graph = DistGraph("test_sampling", part_config=part_config)
+
+    os.environ["DGL_DIST_DEBUG"] = "1"
+
+    edges = {("n3", "r34", "n4"): eids}
+    sampler = dgl.dataloading.MultiLayerNeighborSampler([10, 10], mask="mask")
+    loader = dgl.dataloading.DistEdgeDataLoader(
+        dist_graph, edges, sampler, batch_size=64
+    )
+
+    block = next(iter(loader))[2][0]
+    assert block.num_src_nodes("n1") > 0
+
+
+@pytest.mark.parametrize("num_parts", [1])
+def test_local_masked_sampling_heterograph_gb(
+    num_server,
+):
```

Comment:
If there's only one value do we need to parametrize?

Also, ensure parametrized string name and input argument have the same name

Reply:
Sorry, the code here was a bit rough, but it has been modified now.